### PR TITLE
feat: Added "--scope-case-insensitive" optional argument 

### PR DESCRIPTION
**Pull request by petermESP**
## Content
Added `--scope-case-insensitive` optional argument. Overrides test `['error_scope_capitalization']` with an adjusted regex.

### Using default arguments, only
image

### Using custom arguments
image

### Using default arguments, only
image
### Using `--scope-case-insensitive`
image

## Related
- JIRA task `RDT-793`
- Closes https://github.com/espressif/conventional-precommit-linter/issues/25

### CODE REVIEW

File: conventional_precommit_linter/hook.py [hunk 1]
```diff
@@ -222,6 +231,7 @@ def parse_args(argv: List[str]) -> argparse.Namespace:
     parser.add_argument('--subject-max-length', type=int, default=72, help="Maximum length of the 'Summary'")
     parser.add_argument('--body-max-line-length', type=int, default=100, help="Maximum length of the 'Body' line")
     parser.add_argument('--summary-uppercase', action='store_true', help="'Summary' must start with an uppercase letter")
+    parser.add_argument('--allow-uppercase-in-optional-scope', action='store_true', help='Allow uppercase letters in the optional scope.')
```
Comment (tomassebestik) [inline conventional_precommit_linter/hook.py:234]: This is too long .... can you please change it to something more reasonable (e.g. `--scope-case-insensitive`)

File: conventional_precommit_linter/hook.py [hunk 2]
```diff
@@ -171,9 +174,15 @@ def print_report(commit_type: str, commit_scope: Optional[str], commit_summary:
     rule_messages.append(
         f"{_get_icon_for_rule(rules_output_status['error_scope_format'])} {_color_blue('(<optional-scope>)')} if used, must be enclosed in parentheses"
     )
-    rule_messages.append(
-        f"{_get_icon_for_rule(rules_output_status['error_scope_capitalization'])} {_color_blue('(<optional-scope>)')} if used, must be written in lower case without whitespace"
-    )
+
+    if args.allow_uppercase_in_optional_scope:
+        rule_messages.append(
+            f"{_get_icon_for_rule(rules_output_status['error_scope_capitalization'])} {_color_red('overridden with --allow-uppercase-in-optional-scope')} {_color_blue('(<optional-scope>)')} if used, must be written in lower case without whitespace."
```
Comment (tomassebestik) [inline conventional_precommit_linter/hook.py:180]: This output message is confusing.

1) `overridden with --allow-uppercase-in-optional-scope` part is not necessary and breaks the output format style
2) if the condition changed, then also output message should change

I suggest something like `if used, must be written without whitespace.` for that case.

File: conventional_precommit_linter/hook.py [hunk 3]
```diff
@@ -171,9 +174,15 @@ def print_report(commit_type: str, commit_scope: Optional[str], commit_summary:
...
+        )
+    else:
+        rule_messages.append(
+            f"{_get_icon_for_rule(rules_output_status['error_scope_capitalization'])} {_color_blue('(<optional-scope>)')} if used, must be written in lower case without whitespace. Use '--allow-uppercase-in-optional-scope' to override."
```
Comment (tomassebestik) [inline conventional_precommit_linter/hook.py:184]: Remove `Use '--allow-uppercase-in-optional-scope' to override.`.

The output message for the user should be direct and simple. These extra options are described in the documentation.

**Comment by tomassebestik** [comment]
(Normally, we don't review PRs marked as WIP/Draft. PRs should be in a "ready to merge" state before requesting human reviews.)

This PR is missing a lot of items. Please read the [Contributing Guide](https://github.com/espressif/conventional-precommit-linter/blob/master/CONTRIBUTING.md) and make the necessary fixes:
- Missing tests
- Changes in style without updated thumbnails
- Missing documentation (README.md)
- Missing link to the GitHub issue
- Missing related internal JIRA ticket number; **NOT a link**, just a plain text number like `JIRA-123`

**Review by tomassebestik — CHANGES_REQUESTED** [review]

**Comment by petermESP** [comment]
> (Normally, we don't review PRs marked as WIP/Draft. PRs should be in a "ready to merge" state before requesting human reviews.)
> 
> This PR is missing a lot of items. Please read the [Contributing Guide](https://github.com/espressif/conventional-precommit-linter/blob/master/CONTRIBUTING.md) and make the necessary fixes:
- [x] Missing tests
- [ ] Changes in style without updated thumbnails / thumbnails in PR description
   - @tomassebestik Thumbnails in PR description?
- [x] Missing documentation (README.md)
- [x] Missing link to the GitHub issue
- [x] Missing related internal JIRA ticket number; **NOT a link**, just a plain text number like `JIRA-123`

**Comment by tomassebestik** [comment]
Code looks good, nice job with improving test framework. LGTM

Add please to the commit message body `Closes https://github.com/espressif/conventional-precommit-linter/issues/25`

**Pull request merged**